lib/internal/modules/esm/resolve.js

```diff
@@ -1058,15 +1058,24 @@ function defaultResolve(specifier, context = {}) {
   }
 
   let parsed, protocol;
-  if (shouldBeTreatedAsRelativeOrAbsolutePath(specifier)) {
-    parsed = URLParse(specifier, parsedParentURL);
-  } else {
-    parsed = URLParse(specifier);
-  }
-
-  if (parsed != null) {
```

Comment:
Should it be placed back too?

Reply:
Without parsed != null check, it would fail if URL parsing fails

Reply:
We also don't need the try/catch

Reply:
I think it should be fine now.